base_layer/chat_ffi/src/callback_handler.rs

```diff
@@ -31,8 +32,74 @@ use tari_shutdown::ShutdownSignal;
 
 const LOG_TARGET: &str = "chat_ffi::callback_handler";
 
-pub(crate) type CallbackContactStatusChange = unsafe extern "C" fn(*mut ContactsLivenessData);
-pub(crate) type CallbackMessageReceived = unsafe extern "C" fn(*mut Message);
+pub(crate) type CallbackContactStatusChange = unsafe extern "C" fn(*mut ChatFFIContactsLivenessData);
+pub(crate) type CallbackMessageReceived = unsafe extern "C" fn(*mut ChatFFIMessage);
+
+pub struct ChatFFIContactsLivenessData {
```

Comment:
```suggestion
#[repr(c)]
pub struct ChatFFIContactsLivenessData {
```
if you want to make this available over the FFI you need to add this? If not ignore this message

Reply:
Good catch.